src/runtime_src/xdp/profile/collection/counters.cpp

```diff
@@ -469,25 +469,33 @@ namespace xdp {
 
   }
 
-  // Write host data transfer stats
-  void ProfileCounters::writeHostTransferSummary(ProfileWriterI* writer, bool isRead,
-      uint64_t totalBytes, double totalTimeMsec, double maxTransferRateMBps) const
+  // Write data transfer stats for: host, XDMA, KDMA, and P2P
+  void ProfileCounters::writeTransferSummary(ProfileWriterI* writer, std::string& deviceName,
+      RTUtil::e_monitor_type monitorType, bool isRead, uint64_t totalBytes, uint64_t totalTranx,
+      double totalTimeMsec, double maxTransferRateMBps) const
   {
+    if (monitorType == RTUtil::MON_HOST_DYNAMIC) {
 #ifdef BUFFER_STAT_PER_CONTEXT
-    if (isRead) {
-      for (auto readIter : BufferReadStat)
-        writeBufferStat(writer, "READ", readIter.second, maxTransferRateMBps);
-    }
-    else {
-      for (auto writeIter : BufferWriteStat)
-        writeBufferStat(writer, "WRITE", writeIter.second, maxTransferRateMBps);
-    }
+      if (isRead) {
+        for (auto readIter : BufferReadStat)
```

Comment:
should we use for (const auto& readIter.... or (auto&& readIter.... here to avoid copying of data?

Reply:
Good point. That optional code is not enabled right now. Let's fix it when we actually use it.